### PR TITLE
test dirmanager symlink_or_copy

### LINKED ISSUE #1644: E: golem/resource/dirmanager.py:22:12: No name 'dir_util' in module 'distutils'
https://github.com/golemfactory/golem/blob/66a18964ce43df8c3aa8833c0bf28651611f5ba0/golem/resource/dirmanager.py#L21

### PR DESCRIPTION
**Pull request by etam**
I was supposed to fix #1644, but it seems the import works fine.

It's nice to have more tests anyway.

### CODE REVIEW

File: tests/golem/resource/test_dirmanager.py [hunk 1]
```diff
@@ -2,13 +2,60 @@
...
+        target_path = os.path.join(self.path, 'target')
+
+        with open(source_path, 'w') as f:
+            f.write('source')
+        with open(target_path, 'w') as f:
+            f.write('target')
+
+        def raise_OSError(*_):
+            raise OSError
+        os.symlink = raise_OSError
```
Comment (jiivan) [inline tests/golem/resource/test_dirmanager.py]: This overwrites `os.symlink` for all other tests. Better use `unittest.mock` with `side_effect`.

File: tests/golem/resource/test_dirmanager.py [hunk 2]
```diff
@@ -2,13 +2,60 @@
...
+        source_file_path = os.path.join(source_dir_path, 'file')
+        target_path = os.path.join(self.path, 'target')
+
+        os.mkdir(source_dir_path)
+        with open(source_file_path, 'w') as f:
+            f.write('source')
+
+        def raise_OSError(*_):
+            raise OSError
+        os.symlink = raise_OSError
```
Comment (jiivan) [inline tests/golem/resource/test_dirmanager.py]: `unittest.mock` with side effect

**Review by jiivan — CHANGES_REQUESTED** [review]
Have you also seen pathlib? It may simplify your code (but of course it's just a matter of taste).

File: tests/golem/resource/test_dirmanager.py (hunk 1 above)
Comment (etam) [reply tests/golem/resource/test_dirmanager.py]: I tried
```python
    @patch('os.symlink')
    def test_conflict_file(self, os_symlink):
        os_symlink = Mock(side_effect=OSError)
```
But flake8 says `F841 local variable 'os_symlink' is assigned to but never used` and it seems it doesn't change anything. And I don't want to patch whole `os` module.

Comment (etam) [reply tests/golem/resource/test_dirmanager.py]: I've found a way to patch os.symlink just for tested function.

**Review by jiivan — APPROVED** [review]

**Review by Krigpl — APPROVED** [review]

**Pull request merged**